lib/tooling/base-tool.js

```diff
@@ -24,19 +24,31 @@
 
 import path from 'path';
 
+import PromClient from 'prom-client';
 import _ from 'underscore';
 
 import * as exec from '../exec';
 import {logger} from '../logger';
 import * as utils from '../utils';
 
+const countersPerTool = {};
+
 export class BaseTool {
     constructor(toolInfo, env) {
         this.tool = toolInfo;
         this.env = env;
         this.tool.exclude = this.tool.exclude ? this.tool.exclude.split(':') : [];
         this.addOptionsToToolArgs = true;
         this.parseOutput = utils.parseOutput;
+
+        const toolName = this.tool.name;
+        if (toolName && !(toolName in countersPerTool)) {
+            countersPerTool[toolName] = new PromClient.Counter({
```

Comment:
Pinging @apmorton on this - do you think we should use a single counter with different labels? Else we have no way to do a dashboard with "all tool invocations" without listing all the possible names of tools (IIUC).

Reply:
No worries, it was easy enoug to fix and I already updated them! :) Sorry I should have posted here! #3883